olive/engine/engine.py

```diff
@@ -471,7 +473,7 @@ def run_no_search(
         fp_outputs = self.footprints[accelerator_spec].create_footprints_by_model_ids(output_model_ids)
         # update the output model config
         for model_id, model_config in output_models.items():
-            fp_outputs.nodes[model_id].model_config = model_config
+            fp_outputs.nodes[model_id].model_config = model_config or fp_outputs.nodes[model_id].model_config
```

Comment:
I prefer the following line even if it is same with above line, since
1, the logic is clear
2. no perf penalty. 
```
if model_config:
    fp_outputs.nodes[model_id].model_config = model_config
```

Reply:
https://stackoverflow.com/questions/8747740/assignment-with-or-in-python

Reply:
This is an interesting discussion. We do use `or` in our codebase to check for `None`. 

Both the current line and the proposed line do use the fact that `None` evaluates to False in a boolean expression. But I do see the benefit of point 3. Point 1 is more about if the developer is used to this type of logic or not.

Reply:
Updated.

Reply:
Shared from Mike:
https://stackoverflow.com/questions/21566106/what-does-an-x-y-or-z-assignment-do-in-python

For some easy/readable case, `or` looks pretty clean. I think there is a point looks good:
` final element should probably be a literal`

Reply:
The literal part makes sense if we are defining a new variable like 'x = y or z or default' where we don't know if at least one of x or y is "True"

For the current case, we know that the second argument is always "True" so I don't think it really applies here.